pkg/clustermanager/kubeconfig_manager.go

```diff
@@ -0,0 +1,61 @@
+package clustermanager
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/providers"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+)
+
+type KubeconfigManager struct {
+	ClientFactory ClientFactory
+	// FS is a file system abstraction providing file creation and write capabilities
+	FS filewriter.FileWriter
+}
+
+// UpdateKubeconfig retrieves the contents of the specified cluster's kubeconfig from a secret and copies it to an io.Writer.
+func (km KubeconfigManager) UpdateKubeconfig(ctx context.Context, clusterName, kubeconfig string, provider providers.Provider, w io.Writer) error {
```

Comment:
```suggestion
func (km KubeconfigManager) WriteKubeconfig(ctx context.Context, clusterName, kubeconfig string, provider providers.Provider, w io.Writer) error {
```

Reply:
I would love if we could find a way to not pollute this signature with the provider

The only reason the provider is needed here is because docker needs to make some changes on the fly to the kubeconfig content

What if we just had a different `KubeconfigManager` implementation depending on the provider? That way the "selection" is made at dep build time and we don't need to pass the provider around.